pkg/providers/aws/provider_redis.go

```diff
@@ -255,12 +255,6 @@ func (p *RedisProvider) createElasticacheCluster(ctx context.Context, r *v1alpha
 		}
 		return nil, "started elasticache provision", nil
 	}
-
```

Comment:
I think that moving this check further down would slow down the reporting to the cr of the status being in modifying or maintenance etc.. 🤔  this was an issue for us before in upgrade scenarios.

Reply:
Not sure it would as there is no return between this now and when `foundCache != nil` so should execute in order

Reply:
So if we remove this return we have to wait longer until we get to the cache being nil whereas with it in we get a return quicker if the cache is in any state other than available.

Reply:
discussed off line